SalesforceSDK/Core/Auth/OAuth2.cs

```diff
@@ -323,30 +298,53 @@ public static string ComputeFrontDoorUrl(string instanceUrl, string displayType,
         /// <returns>Boolean based on if the refresh auth token succeeded or not</returns>
         public static async Task<Account> RefreshAuthToken(Account account)
         {
-            if (account != null)
+            LoggingService.Log("Atempting to refresh auth token", LoggingLevel.Verbose);
+
+            if (account == null)
             {
-                try
-                {
-                    AuthResponse response =
-                        await RefreshAuthTokenRequest(account.GetLoginOptions(), account.RefreshToken);
-                    account.AccessToken = response.AccessToken;
-                    await SDKServiceLocator.Get<IAuthHelper>().PersistCredentialsAsync(account);
-                }
-                catch (WebException ex)
-                {
-                    LoggingService.Log("Exception occurred when refreshing token:", LoggingLevel.Critical);
-                    LoggingService.Log(ex, LoggingLevel.Critical);
-                    Debug.WriteLine("Error refreshing token");
-                    throw new OAuthException(ex.Message, ex.Status);
-                }
-                catch (Exception ex)
-                {
-                    LoggingService.Log("Exception occurred when refreshing token:", LoggingLevel.Critical);
-                    LoggingService.Log(ex, LoggingLevel.Critical);
-                    Debug.WriteLine("Error refreshing token");
-                    throw new OAuthException(ex.Message, ex.InnerException);
-                }
+                return null;
+            }
+
+            try
+            {
+                var loginOptions = account.GetLoginOptions();
+
+                // args
+                var argsStr = string.Format(OauthRefreshQueryString, loginOptions.ClientId, account.RefreshToken);
+
+                // Refresh url
+                var refreshUrl = loginOptions.LoginUrl + OauthRefreshPath;
+
+                // Post
+                var call = HttpCall.CreatePost(refreshUrl, argsStr);
+
+                var response = await call.ExecuteAndDeserialize<AuthResponse>();
+
+                account.AccessToken = response.AccessToken;
+
+                await SDKServiceLocator.Get<IAuthHelper>().PersistCredentialsAsync(account);
+            }
+            catch (DeviceOfflineException ex)
```

Comment:
wrap all non-connectivity issues in a oauth exception so the consumer knows that we failed to update the oauth token for a legitimate reason (not because we were offline)